.github/workflows/build.yml

```diff
@@ -0,0 +1,214 @@
+name: build
+
+# Controls when the action will run
+on:
+
+  # Trigger the workflow on all pushes, except on tag creation
+  push:
+    branches:
+    - '**'
+    tags-ignore:
+    - '**'
+
+  # Trigger the workflow on all pull requests
+  pull_request: ~
+
+  # Allow workflow to be dispatched on demand
+  workflow_dispatch: ~
+
+env:
+  FIELD_API_TOOLS: ${{ github.workspace }}/.github/tools
+  CTEST_PARALLEL_LEVEL: 1
+  CACHE_SUFFIX: v0         # Increase to force new cache to be created
+  DEV_ALLOC_SIZE: 1000000
+
+jobs:
+  ci:
+    name: ci
+
+    strategy:
+      fail-fast: false  # false: try to complete all jobs
+
+      matrix:
+        build_type: [RelWithDebInfo]
+        name:
+          - linux gnu-10
+          - linux gnu-14
+          - linux nvhpc-23.5
+          - linux intel-classic
+          - linux intel-modern
+          - macos
+
+        include:
+
+          - name: linux gnu-10
+            os: ubuntu-20.04
```

Comment:
Any particular reason for using the relatively old runner image here? 20.04 is EOL in a couple of months and I expect GH might make these runners unavailable soon after.

Reply:
Great catch!